### PR TITLE
Clamp "Expires" to 9999 years

### PR DESCRIPTION
**Pull request by erickt**
According to [rfc 6265](https://tools.ietf.org/html/rfc6265#section-5.1.1), cookies support at most 9999 years.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -675,7 +675,11 @@ impl<'c> Cookie<'c> {
     /// # }
     /// ```
     #[inline]
-    pub fn set_expires(&mut self, time: Tm) {
+    pub fn set_expires(&mut self, mut time: Tm) {
+        // According to https://tools.ietf.org/html/rfc6265#section-5.1.1, cookie expires needs 
+        // to be less than or equal to 9999 years.
+        if time.tm_year > 8099 { time.tm_year = 8099; }
```
Comment (SergioBenitez) [inline src/lib.rs]: Is it `8099` or `9999`? Are you accounting for the possibility that a `Tm` may not make sense, i.e, where `month` might be 2^16 or something nonsensical?

**Comment by SergioBenitez** [comment]
There's a build failure.

File: src/lib.rs (hunk 1 above)
Comment (erickt) [inline src/lib.rs]: I'll add a comment. `Tm.tm_year` starts at 1900, so the max value we can have is 8099 + 1900 = 9999.

**Comment by SergioBenitez** [comment]
Merged in https://github.com/SergioBenitez/cookie-rs/commit/121d6887131e52b38cdf0b39a067c25651f89ed5.

**Pull request closed without merging**